HeterogeneousCore/AlpakaInterface/interface/VecArray.h

```diff
@@ -0,0 +1,107 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_VecArray_h
+#define HeterogeneousCore_AlpakaInterface_interface_VecArray_h
+
+//
+// Author: Felice Pantaleo, CERN
+//
+
+#include <utility>
+
+#include <alpaka/alpaka.hpp>
+
+namespace cms::alpakatools {
+
+  template <class T, int maxSize>
+  class VecArray {
+  public:
+    using self = VecArray<T, maxSize>;
+    using value_t = T;
+
+    inline constexpr int push_back_unsafe(const T &element) {
+      auto previousSize = m_size;
+      m_size++;
+      if (previousSize < maxSize) {
+        m_data[previousSize] = element;
+        return previousSize;
+      } else {
+        --m_size;
+        return -1;
+      }
+    }
+
+    template <class... Ts>
+    constexpr int emplace_back_unsafe(Ts &&...args) {
+      auto previousSize = m_size;
+      m_size++;
+      if (previousSize < maxSize) {
+        (new (&m_data[previousSize]) T(std::forward<Ts>(args)...));
+        return previousSize;
+      } else {
+        --m_size;
+        return -1;
+      }
+    }
+
+    inline constexpr T &back() const {
```

Comment:
```suggestion
    inline constexpr T const& back() const {
```
?

Reply:
Indeed. (I will add the mutable version too)